tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -182,9 +200,16 @@ def _ValidateTranslation(self):
             print('Invalid translation unit', unit)
             return False
       else:
-        print('Translation has improper keys:', device_name)
+        print('Translation has improper keys:', field_name)
         return False
 
+    #check if the rest of the fields not included are optional
+    for optional_field_name in all_fields_dict.values():
+      if not optional_field_name.optional:
```

Comment:
doesn't this just fail any time a type has an optional field?

Reply:
no the opposite

Reply:
if it is not optional it will return